### PR TITLE
Added expand() method for SeqFormula

### LINKED ISSUE #12634: expand() doesn't work with SeqFormula
Sympy 1.0-2 from Manjaro repository.

```
In [1]: import sympy as sp

In [2]: x, y = sp.symbols('x, y')

In [3]: i = sp.symbols('i', integer=True)

In [4]: seq = sp.SeqFormula(x * (y + i), (i, 1, 5))

In [5]: sp.expand(seq)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-5-b6d735e12974> in <module>()
----> 1 sp.expand(seq)

/usr/lib/python3.6/site-packages/sympy/core/function.py in expand(e, deep, modulus, power_base, power_exp, mul, log, multinomial, basic, **hints)
   2019     hints['multinomial'] = multinomial
   2020     hints['basic'] = basic
-> 2021     return sympify(e).expand(deep=deep, modulus=modulus, **hints)
   2022 
   2023 # This is a special application of two hints

AttributeError: 'SeqFormula' object has no attribute 'expand'
```

### PR DESCRIPTION
**Pull request by pragyanmehrotra**
#### References to other Issues or PRs

Fixes #12634 

#### Brief description of what is fixed or changed
Now we can get an expansion of the sequence formula,
By doing-
```python
>>> x = Symbol('x')
>>> y = Symbol('y', integer = True)
>>> seq = SeqFormula(sin(x*(x+y)), (y,1,5))
>>>expand(seq)
SeqFormula(sin(x**2)*cos(x*y) + sin(x*y)*cos(x**2), (y, 1, 5))
>>> seq.expand()
SeqFormula(sin(x**2)*cos(x*y) + sin(x*y)*cos(x**2), (y, 1, 5))
```
#### Release Notes

* series
    * Added expand method for SeqFormula

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
Isn't SeqFormula a sequence? What does it mean that you expanded it and only ended up with a single term?

Should it not return a new SeqFormula?

**Comment by pragyanmehrotra** [comment]
Oh right! I should convert it to SeqFormula at the end.

**Comment by pragyanmehrotra** [comment]
Done, please have a look at it.

**Comment by oscarbenjamin** [comment]
Seems reasonable. I think this is a good case for using `*args, **kwargs` since you are just passing almost all arguments through.

This will need tests.

**Comment by oscarbenjamin** [comment]
Is there any possibility that the limits of the SeqFormula should be expanded? E.g.:
```
SeqFormula(sin(x), (x, 0, y*(y+1)))  
```

**Comment by pragyanmehrotra** [comment]
Hmmm, that could be expanded as well. Should I create an optional parameter for it?

**Comment by pragyanmehrotra** [comment]
and where should I add the tests? @oscarbenjamin

**Comment by Naba7** [comment]
in the tests folder under sympy/series

**Comment by oscarbenjamin** [comment]
It's probably best not to bother with expanding the limits unless someone has a use for it.

I notice that you can't even print that kind of object:
```
In [11]: pprint(s)                                                                                       
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
...
NotImplementedError: Pretty printing of sequences with symbolic bound not implemented
```

Not sure how it should print. Perhaps:
```
\left{sin(x)\right}_{x=0}^{y(y+1)}
```

**Comment by oscargus** [comment]
I think it prints like that in Latex at least.

File: sympy/series/sequences.py [hunk 1]
```diff
@@ -713,6 +713,11 @@ def coeff_mul(self, coeff):
         formula = self.formula * coeff
         return SeqFormula(formula, self.args[1])
 
+    def expand(self, deep = True, modulus = None, mul = True, multinomial = True, basic = True,
```
Comment (oscargus) [inline sympy/series/sequences.py]: Please remove the spaces around = for optional parameters.

File: sympy/series/sequences.py [hunk 2]
```diff
@@ -713,6 +713,11 @@ def coeff_mul(self, coeff):
...
+     trig = True, log = True, power_exp = True, power_base = True):
+
+        return SeqFormula(expand(self.formula, deep = deep, modulus = modulus, mul = mul, trig = trig, log = log, multinomial = multinomial,
```
Comment (oscargus) [inline sympy/series/sequences.py]: Same here.

File: sympy/series/tests/test_sequences.py [hunk 3]
```diff
@@ -75,6 +75,15 @@ def test_SeqFormula():
     raises(ValueError, lambda: SeqFormula(n**2, (n, -oo, oo)))
     raises(ValueError, lambda: SeqFormula(m*n**2, (0, oo)))
 
+    seq = SeqFormula(x*(y**2 + z), (z, 1, 100))
+    assert seq.expand() == SeqFormula(x*y**2 + x*z, (z, 1, 100))
+    seq = SeqFormula(sin(x*(y**2 + z)),(z, 1, 100))
+    assert seq.expand() == SeqFormula(sin(x*y**2)*cos(x*z) + sin(x*z)*cos(x*y**2), (z, 1, 100))
+    assert seq.expand(trig = False) == SeqFormula(sin(x*y**2 + x*z), (z, 1, 100))
```
Comment (oscargus) [inline sympy/series/tests/test_sequences.py]: And here (and below).

**Comment by oscargus** [comment]
Some minor comments, but indeed @oscarbenjamin's comment about using `*args, **kwargs` also makes sense (at least for these "general" methods). Still, my comments hold for the formatting of test inputs.

**Comment by pragyanmehrotra** [comment]
What do you mean by using ` *args, **kwargs ` ? Is some change needed in ` self.args[1] ` ?

**Comment by oscargus** [comment]
You can write the function as 
```
    def expand(self, *args, **kwargs):
        return SeqFormula(expand(self.formula, *args, **kwargs), self.args[1])
```
In that way all unnamed arguments (none here) will end up in a list(?) `args` and all named arguments in a dict `kwargs`. Then all are passed to the next call (*args means that they are not passed as a list, but as individual arguments, similar with **kwargs).

Sometimes it is better to use named arguments in the call, sometimes not... I think that here as it is a method that applies to a lot of different classes it can make sense to just use the "simple" version with *arg, **kwargs. For a dedicated method it is probably more useful to name them in the method call.

File: sympy/series/tests/test_sequences.py [hunk 4]
```diff
@@ -78,10 +78,10 @@ def test_SeqFormula():
     seq = SeqFormula(x*(y**2 + z), (z, 1, 100))
     assert seq.expand() == SeqFormula(x*y**2 + x*z, (z, 1, 100))
     seq = SeqFormula(sin(x*(y**2 + z)),(z, 1, 100))
-    assert seq.expand() == SeqFormula(sin(x*y**2)*cos(x*z) + sin(x*z)*cos(x*y**2), (z, 1, 100))
```
Comment (oscargus) [inline sympy/series/tests/test_sequences.py]: I think it makes sense to include the default behavior (without additional arguments) in the test as well (but good that you added the trig=True case).

Same thing for power_exp below.

Comment (pragyanmehrotra) [reply sympy/series/tests/test_sequences.py]: I think the default behavior is when `trig=False`

Comment (pragyanmehrotra) [reply sympy/series/tests/test_sequences.py]: I should probably get rid of that then

Comment (oscargus) [reply sympy/series/tests/test_sequences.py]: No, please keep both. There is a purpose to test both the behaviour of `trig=False` and that `trig=False` indeed is the default behaviour.

Comment (pragyanmehrotra) [reply sympy/series/tests/test_sequences.py]: Done

**Comment by oscargus** [comment]
Thanks!

**Pull request merged**